### ISSUE TITLE
TMSElevationLayer->setURL(...) causes error...

### ISSUE THREAD

**Issue by keince** [gwaldron/osgearth#2160]
Recently, our group purchased the `readymap` data (both elevation and imagery) from Pelican Mapping. The data was sent on a hard-drive for us to use internally.

We are able to load the data without any trouble using the provided `readymap3.earth` file. However, when we try to load the each layer programmatically using the following code, we run into problems...

`TMSImageLayer* rmImagery = new TMSImageLayer();`
`rmImagery->setName("readymap_imagery");`
`rmImagery->setURL("X:/osgEarthData/tiles/readymap_imagery/");`
`rmImagery->setFormat("jpeg");`
`rmImagery->setProfile(Profile::create(Profile::GLOBAL_GEODETIC));`
`mapNode->getMap()->addLayer(rmImagery);`

`TMSElevationLayer* rmElevation = new TMSElevationLayer();`
`rmElevation->setName("readymap_elevation");`
`rmElevation->setURL("X:/osgEarthData/tiles/readymap_elevation/");`
`rmElevation->setFormat("tif");`
`rmElevation->setProfile(Profile::create(Profile::GLOBAL_GEODETIC));`
`rmElevation->setTileSize(257);`
`mapNode->getMap()->addLayer(rmElevation);`

When we run the program, the TMS elevation layer is throwing the following error...

**TMS driver requires a valid "url" property**

The TMS image layer appears to not have any problem with its URL which is puzzling since the URLs are pretty much at the same location. Not sure if this is a bug in osgEarth or improper usage on our part...

**Comment by gwaldron** [comment]
Hello,
For TMS in the filesystem, try pointing your URL at the actual tms.xml file; for example:
```c++
rmImagery->setURL("X:/osgEarthData/tiles/readymap_imagery/tms.xml");
```
There is no need to set format, profile, tile size, etc. These are all "hard wired" when using TMS.
Hope this helps!

**Comment by keince** [comment]
Glenn,

Thanks for the response. 

Unfortunately, your suggestion of referencing the `tms.xml` directly will not work for our use case because we don't have access to the internet. The path references in that xml file such as `http://readymap.org/readymap/tiles/1.0.0/116/0` obviously require internet access. We purchased the readymap data directly on a hard-drive from Pelican Mapping so we can reference the data in a standalone environment with no connectivity to the internet.

Regarding your second comment, we set the format, profile, tile size programmatically because that's what the `readymap3.earth` file that we received was doing. Per your suggestion, we can certainly ignore those things but I doubt that's the source of our problems in not being able to load the `TMSElevationLayer` because of problems with the URL I described above. Again, the `TMSImageLayer` loads the data just fine with its URL. The `TMSElevationLayer` doesn't.

It appears to be a bug somewhere in the TMS driver when specifying an absolute URL for the elevation layer programmatically. 

Shayne
